enterprise/reporting/event_stats.go

```diff
@@ -0,0 +1,52 @@
+package reporting
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+const (
+	EventsDeliveredMetricName = "events_delivered_total"
+	EventsAbortedMetricName   = "events_aborted_total"
+)
+
+type EventStatsReporter struct {
+	stats            stats.Stats
+	configSubscriber *configSubscriber
+}
+
+func NewEventStatsReporter(configSubscriber *configSubscriber, stats stats.Stats) *EventStatsReporter {
+	return &EventStatsReporter{
+		stats:            stats,
+		configSubscriber: configSubscriber,
+	}
+}
+
+func (es *EventStatsReporter) Report(metrics []*types.PUReportedMetric) error {
+	// tracking delivery event stats which has state - succeeded, aborted
+	for _, metric := range metrics {
+		metric := metric
+		if !(metric.StatusDetail.Status == "aborted" || metric.StatusDetail.Status == "succeeded") {
```

Comment:
we also have filtered and migrated terminal statuses. Do we want to also send stats for these states?

Reply:
discussed offline, not needed to capture these states for the alerting use case